cdmtaskservice/s3/remote.py

```diff
@@ -234,6 +242,66 @@ async def _process_uploads(
     await _run_tasks(tasks, concurrency)
 
 
+# TODO REFDATA clean up tar and gz files
+def _ensure_safe_tar_path(parent_dir: Path, member_name: str, tar_file: str):
+    """Ensure that the target path is within the base directory."""
+    # TODO TEST need to figure out how to make a bad tar file to test this
+    abs_base = parent_dir.absolute()
+    abs_target = (parent_dir / member_name).absolute()
+    if not abs_base.parts == abs_target.parts[:len(abs_base.parts)]:
+        raise ValueError(
+            f"Unsafe path detected for tarfile {tar_file}: {member_name}"
+        )
+
+
+def _extract_gz(file_path: Path):
+    """Extract a .gz file."""
+    output_path = file_path.parent / file_path.stem
+    with gzip.open(file_path, 'rb') as f_in, open(output_path, 'wb') as f_out:
+        shutil.copyfileobj(f_in, f_out)
+
+
+def _extract_tar(file_path: Path):
+    """Extract a .tar.gz or .tgz file safely."""
+    parent_dir = file_path.parent
+    with tarfile.open(file_path, 'r:*') as tar:
+        for member in tar.getmembers():
+            _ensure_safe_tar_path(parent_dir, member.name, file_path)
+        tar.extractall(parent_dir)
+
+
+async def _process_download(
+    session: aiohttp.ClientSession,
+    url: str,
+    partsize: int,
+    outputpath: Path,
+    etag: str,
+    insecure_ssl: bool,
+    timeout_sec: int,
+    unpack: bool,
+):
+    await download_presigned_url(
+        session,
+        url,
+        partsize,
+        outputpath,
+        etag,
+        insecure_ssl,
+        timeout_sec
+    )
+    if unpack:
+        op = str(outputpath)
+        if op.endswith(_EXT_TGZ) or op.endswith(_EXT_TARGZ):
```

Comment:
```suggestion
        if op.endswith(_EXT_TARGZ) or op.endswith(_EXT_TGZ):
```

Reply:
What's the reasoning for this change? I'm not following

Reply:
the original logic checks `.gz` then `.tar.gz`. But `endswith ('.tar.gz')` will always the same as `endswith ('.gz' )` right?

Reply:
No, the original logic checks `.tgz` then `.tar.gz`...?

Reply:
oh oh. I read it wrong. Never mind.